terraform/resource_provider.go

```diff
@@ -237,13 +237,11 @@ func ResourceProviderResolverFixed(factories map[addrs.Provider]ResourceProvider
 		ret := make(map[addrs.Provider]ResourceProviderFactory, len(reqd))
 		var errs []error
 		for name := range reqd {
-			// FIXME: discovery.PluginRequirements should use addrs.Provider as
-			// the map keys instead of a string
-			fqn := addrs.NewLegacyProvider(name)
+			fqn, _ := addrs.ParseProviderSourceString(name)
```

Comment:
This is intended as a short-term shim, but useful as long as we're still using the generic plugin installer (I can't just force it to use `addrs.Provider` as map keys when it could also be installing provisioners). If somehow we do end up using fqn strings as map keys I will replace this with a slightly different function meant only to de-serialize fqn strings (I would rather not do this thing, but we'll see what reality has to say about it).